include/amy/basic_connector.hpp

```diff
@@ -95,22 +95,15 @@ class basic_connector : public AMY_ASIO_NS::basic_io_object<MySQLService> {
     }
 
     template<typename Endpoint, typename ConnectHandler>
-    BOOST_ASIO_INITFN_RESULT_TYPE(ConnectHandler,
-                                  void (AMY_SYSTEM_NS::error_code))
-    async_connect(Endpoint const& endpoint,
```

Comment:
Just curious, I'm not quite familiar with recent versions of (Boost.)Asio. Is `BOOST_ASIO_INITFN_RESULT_TYPE` meant for enabling using asynchronous functions together with coroutines?